editor.planx.uk/src/@planx/components/MultipleFileUpload/Public.tsx

```diff
@@ -40,52 +45,52 @@ const DropzoneContainer = styled(Box)(({ theme }) => ({
   },
 }));
 
-const slotsSchema = array()
-  .required()
-  .test({
-    name: "nonUploading",
-    message: "Upload at least one file",
-    test: (slots?: Array<FileUploadSlot>) => {
-      return Boolean(
-        slots &&
-          slots.length > 0 &&
-          !slots.some((slot) => slot.status === "uploading")
-      );
-    },
+function Component(props: Props) {
+  const [fileList, setFileList] = useState<FileList>({
+    required: [],
+    recommended: [],
+    optional: [],
   });
 
-function Component(props: Props) {
-  const recoveredSlots = getPreviouslySubmittedData(props)?.map(
-    (slot: FileUploadSlot) => slot.cachedSlot
-  );
-  const [slots, setSlots] = useState<FileUploadSlot[]>(recoveredSlots ?? []);
+  useEffect(() => {
+    const passport = useStore.getState().computePassport();
+    const fileList = createFileList({ passport, fileTypes: props.fileTypes });
+    setFileList(fileList);
+  }, []);
+
+  useEffect(() => {
+    // TODO: Re-map slots to userfiles also?
+    const recoveredSlots: FileUploadSlot[] = getRecoveredSlots(
+      props.previouslySubmittedData,
+      fileList
+    );
+    setSlots(recoveredSlots);
+  }, [props.previouslySubmittedData, fileList]);
+
+  const [slots, setSlots] = useState<FileUploadSlot[]>([]);
+
   const [fileUploadStatus, setFileUploadStatus] = useState<string | undefined>(
     undefined
   );
+
   const [validationError, setValidationError] = useState<string | undefined>();
   const [showModal, setShowModal] = useState<boolean>(false);
 
   const handleSubmit = () => {
-    slotsSchema
-      .validate(slots)
+    // This is a temp cheat to bypass tagging
+    // The first slot is mapped to a single required file
+
+    // const fileListWithTaggedFile = {...fileList}
+    // fileListWithTaggedFile.required[0].slot = slots[0]
+    // setFileList(fileListWithTaggedFile);
+
+    Promise.all([
+      slotsSchema.validate(slots),
```

Comment:
One thing I'm not 100% certain on just now is how we validate that all uploaded files are tagged. I expect this will be part of the modal logic, but it might be nice at this stage to validate that the length of `slots` is equal to the filtered `fileList`. This is something I'll come back to shortly once the modal tagging logic is in place.